src/animation/event.rs

```diff
@@ -0,0 +1,95 @@
+use bevy::prelude::*;
+
+use crate::asset::AnimationClip2D;
+
+use super::{
+    animation_spritesheet::animation_player_spritesheet, AnimationPlayer2D,
+    AnimationPlayer2DSystemSet,
+};
+
+/// AnimationEvents are triggered by the animation system if registered as such with the App.
+pub trait AnimationEvent: Event + FromReflect {
+    /// Implement this to be able to set the entity for a targeted event.
+    /// Default implementation is a No-Op.
+    fn set_entity(&mut self, entity: Entity) {
+        let _ = entity;
+        /* Default implementation is empty for non-targeted events */
+    }
+}
+
+pub fn default_entity() -> Entity {
+    Entity::PLACEHOLDER
+}
+
+// Define generic trigger and event systems
+pub fn send_animation_event<T: AnimationEvent>(
+    mut event_writer: EventWriter<T>,
+    animation_players: Query<(Entity, &AnimationPlayer2D)>,
+    animation_clips: Res<Assets<AnimationClip2D>>,
+) {
+    for (entity, animation_player) in &animation_players {
+        if let Some(animation_clip) = animation_clips.get(animation_player.animation_clip()) {
+            if let Some(reflected_events) = animation_clip.events().get(&animation_player.frame()) {
+                for reflected_event in reflected_events {
+                    // TODO: Patch in entity somehow
+                    if let Some(mut event) = T::from_reflect(reflected_event.as_reflect()) {
```

Comment:
Quick reminder for myself to check whether there might be a better method to check if  FromReflect for the given T is possible.